### ISSUE TITLE
Semantics and Expand/Flex don't play well together

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#15207]
You cannot wrap an Expand widget in a Semantics widget to give a semantics node the full size of the expand widget.

**Comment by Hixie** [comment]
Expand doesn't actually correspond to a render object, it just configures its parent to make its child expanded. Does it work if you put the Semantics as the child of the Expanded?

**Comment by goderbauer** [comment]
I meant to update this: yeah, that actually works and that's what I ended up doing. I'll close this for now. We should find a good place to document this, though.

**Comment by tarobins** [comment]
I seem to be running into a variant of this issue.  I have widget A with a number of decendents, including an Expanded a few levels down, but I want treat the widget A and all it's decendants as one Semantic unit.  When I add either a Semantics or a MergeSemantics around A, then I get the exception 

```
[04061.476][01112][01185][g3_assistant_aot_product_runner] INFO: opal.cm(flutter): type 'ParentData' is not a subtype of type 'FlexParentData' in type cast
[04061.482][896012][909036][kronk] INFO: [speech/audio/processor/playout_eraser.cc:419] OnEraserStateChange: Passing-through. -> Aligning
[04061.483][01112][01178][g3_assistant_aot_product_runner] INFO: opal.cm(flutter): #0      Flexible.applyParentData (package:flutter/src/widgets/basic.dart:5196)
[04061.483][01112][01178][g3_assistant_aot_product_runner] INFO: opal.cm(flutter): #1      RenderObjectElement._updateParentData (package:flutter/src/widgets/framework.dart:6249)
[04061.483][01112][01178][g3_assistant_aot_product_runner] INFO: opal.cm(flutter): #2      RenderObjectElement.attachRenderObject (package:flutter/src/widgets/framework.dart:6270)
[04061.483][01112][01178][g3_assistant_aot_product_runner] INFO: opal.cm(flutter): #3      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6154)
```

How do I group a subtree containing an Expanded into one sematic unit?

**Comment by tarobins** [comment]
I eventually got this to work.  I think there was a Postioned.fill which was actually the source of the problem, but semantics and Flex/Positioned/??  seems like a delicate thing.
